### PR TITLE
feat(init): Dont start networkd when running in a container

### PR DESCRIPTION
**Pull request by bradbeam**
Signed-off-by: Brad Beam <brad.beam@talos-systems.com>

### CODE REVIEW

File: pkg/userdata/userdata.go [hunk 1]
```diff
@@ -28,12 +28,13 @@ import (
 
 // UserData represents the user data.
 type UserData struct {
+	Debug      bool        `yaml:"debug"`
```
Comment (bradbeam) [inline pkg/userdata/userdata.go]: Linter was yelling at me for the maligned struct

File: internal/app/init/main.go [hunk 2]
```diff
@@ -77,6 +77,7 @@ func container() (err error) {
 		if decoded, err = base64.StdEncoding.DecodeString(*userdataArg); err != nil {
 			return err
 		}
+		decoded = append(decoded, []byte("\ncontainer: true")...)
```
Comment (andrewrynhard) [inline internal/app/init/main.go]: I think we should just set this after we do:
```
	var data *userdata.UserData
	if data, err = userdata.Open(constants.UserDataPath); err != nil {
		return err
	}
```

a few lines down.

```
data.Container = true
```

**Comment by smira** [comment]
I don't quite like skipping networkd for containers as it "hides" it from CICD, but if we are to skip, the easiest way is to return `nil` runner from `Runner()` method of the service, that marks service as 'Skipped' and that also sends out 'up' event, so we can keep dependencies same

**Comment by smira** [comment]
(I mean returning nil based on userData from `func (c *Networkd) Runner(...)` in `system/services/networkd.go`

**Comment by andrewrynhard** [comment]
Ahh yes we have a `StateSkipped`. I think that is the better approach.

**Comment by andrewrynhard** [comment]
@bradbeam You plan on a follow up PR?

**Comment by bradbeam** [comment]
Negative, sounded like it wasn't necessary.

**Pull request closed without merging**